spec/controllers/boxes_controller_spec.rb

```diff
@@ -267,18 +267,17 @@
       end.to change(institution.boxes, :count).by(0)
     end
 
-    def expect_samples(batch, concentration_number: 1, concentration_exponents: [0], replicates:)
+    def expect_samples(batch, concentrations: [1], replicates:)
       # NOTE: can't where/order in SQL because entity fields...
-      samples = batch.samples.to_a.reject { |s| s.concentration_number != concentration_number }.sort! do |a, b|
-        cmp = a.concentration_exponent <=> b.concentration_exponent
-        cmp = a.replicate <=> b.replicate if cmp == 0
-        cmp
-      end
+      concentrations.each do |c|
+        samples = batch.samples.to_a.reject { |s| s.concentration != c }.sort! do |a, b|
+          cmp = a.replicate <=> b.replicate
+          cmp
```

Comment:
Let's simplify a bit:

```suggestion
        samples = batch.samples.to_a
          .reject { |s| s.concentration != c }
          .sort_by(&:replicate)
```